### PR TITLE
[Common Lisp] Implement new Concept Exercise: conditionals

### LINKED ISSUE #1012: [Common Lisp] Implement new Concept Exercise: conditionals
This document describes how to implement the `conditionals` concept exercise for the Common Lisp track.

## Getting started

**Please please please read the docs before starting.** Posting PRs without reading these docs will be a lot more frustrating for you during the review cycle, and exhaust Exercism's maintainers' time. So, before diving into the implementation, please read up on the following documents:

- [The features of v3](https://github.com/exercism/v3/blob/master/docs/concept-exercises.md).
- [Rationale for v3](https://github.com/exercism/v3/blob/master/docs/rationale-for-v3.md).
- [What are concept exercise and how they are structured?](https://github.com/exercism/v3/blob/master/docs/features-of-v3.md)

Please also watch the following video:

- [The Anatomy of a Concept Exercise](https://www.youtube.com/watch?v=gkbBqd7hPrA).

## Goal

The goal of this exercise is to teach the student about the various `conditionals` within Common Lisp. The student should be able to recognize which conditional is the most appropriate in a given context and be able to confidently write `when`, `unless`, `if`, `cond`, and `case` expressions.

Additionally, the student will be introduced to the concept of "true" and "false" values. That is to say, in Common Lisp, `nil` and `'()` are "false", while all other values are "true"

## Learning objectives

- Know how to use `when` and `unless` for single-branch conditions
- Know how to use `if` for two-branch conditions
- Know how to use `cond` for multi-branch conditions
- Know how to use `case` and `ecase` for special, multi-branch conditions
- Understand which values are "false" and which are "true"

## Out of scope

- Sameness / Equality (`eq`, `eql`, `equal`, `equalp`, etc.)
- Short-circuit evaluation with `and` and `or`
- Conditions and restarts – those are *very* different ;)
- The `ccase` macro, which actually does involve conditions and restarts
- The type-matching `typecase` macro
- Any non-conditional or compound form of control flow

## Concepts

- `conditionals`
- `truthy-and-falsy`

## Prerequisites

- `expressions`
- `integers`
- `arithmetic`
- `symbols`

## Resources to refer to

- [Riptutorial](https://riptutorial.com/common-lisp/example/11082/conditional-constructs)
- [Common Lisp the Language](https://www.cs.cmu.edu/Groups/AI/html/cltl/clm/node84.html)

### Hints

- TBD

### After

- TBD

## Representer

- Currently, no extra logic needs to be added to the [representer](https://github.com/exercism/common-lisp-representer).

## Analyzer

- The [analyzer](https://github.com/exercism/common-lisp-analyzer) would benefit from the ability to check which conditionals are employed by the student. It's often not possible to test for `cond` vs `if` vs `when`, so the analyzer, not the test suite, needs to pick up on this.

## Implementing

When implementing this exercise, avoid the topic of `sameness` as much as possible, as that is its own concept. Stick to only using values as true / false or basic numerical relations (`<`, `>=`, `=`, etc.) and avoid `eq`, `eql`, `equal`, etc.

The `case` macro is a bit of an exception here. Just be sure to use symbols or numbers here so that the default `:test` does not need to be changed from `eql`.

## Help

If you have any questions while implementing the exercise, please post the questions as comments in this issue.

### PR DESCRIPTION
**Pull request by TheLostLambda**
Eventually closes #1012, just a tracking PR for now.

### CODE REVIEW

File: languages/common-lisp/exercises/concept/conditionals/.meta/design.md [hunk 1]
```diff
@@ -0,0 +1,29 @@
+## Learning objectives
+
+- Know how to use `when` and `unless` for single-branch conditions
+- Know how to use `if` for two-branch conditions
+- Know how to use `cond` for multi-branch conditions
+- Know how to use `case` and `ecase` for special, multi-branch conditions
```
[the remark this answers is not in the document]
Comment (TheLostLambda) [reply languages/common-lisp/exercises/concept/conditionals/.meta/design.md:6]: It's currently under "Out of scope", but we can certainly move it :)

[the remark this answers is not in the document]
Comment (verdammelt) [reply languages/common-lisp/exercises/concept/conditionals/.meta/design.md:6]: ah. hmmm not important enough to me. Leave it out.

**Comment by TheLostLambda** [comment]
Thanks so much to both @mikedamay and @verdammelt for all of the help on this one! I think I'm feeling about ready to merge things :slightly_smiling_face: I've made a couple Issues for improving exercises leading up to this and we'll address the docstring thing once we've introduced the concept to the student (either in `basics` or `function-definition`).

I'll leave this open for a day or so, so that people have time to get in any last words :slightly_smiling_face:

File: languages/common-lisp/exercises/concept/conditionals/.docs/after.md [hunk 2]
```diff
@@ -0,0 +1,164 @@
+## Summary
```
Comment (ErikSchierboom) [inline languages/common-lisp/exercises/concept/conditionals/.docs/after.md:1]: This document currently does not contain link to external resources. Are there maybe resources to link to?

File: languages/common-lisp/exercises/concept/conditionals/.docs/instructions.md [hunk 3]
```diff
@@ -0,0 +1,88 @@
+When the Lisp aliens first came to Earth, they encountered countless weird and
```
Comment (ErikSchierboom) [inline languages/common-lisp/exercises/concept/conditionals/.docs/instructions.md:1]: Love the story!

File: languages/common-lisp/exercises/concept/conditionals/.docs/instructions.md [hunk 4]
```diff
@@ -0,0 +1,88 @@
...
+
+## 1. Picking a Pal
+
+Though Ludwig is set on getting a pet, that's about as far as he's gotten in
+thinking things through. The first step, then, is deciding what sort of pet to
+get!
+
+Ludwig wants to find a pal with a personality that complements his own. Could
+you write Ludwig a function that takes some personality trait (a keyword) and
+returns the type of pet (a string) with the given trait?
```
Comment (ErikSchierboom) [inline languages/common-lisp/exercises/concept/conditionals/.docs/instructions.md:18]: Could you maybe explicitly mention the function name to implement (`pal-picker`) in this task description?

File: languages/common-lisp/exercises/concept/conditionals/.docs/instructions.md [hunk 5]
```diff
@@ -0,0 +1,88 @@
...
+(pal-picker :quiet) ; => "Fish"
+```
+
+## 2. In Their Natural Habitat
+
+Now that Ludwig has a new friend, they'll need a place to stay! Can you help
+Ludwig select the right size bed / tank / cage for his pet?
+
+Here, Ludwig needs a function for selecting the proper habitat size (a keyword)
+from his pet's weight in kilograms (a integer).
```
Comment (ErikSchierboom) [inline languages/common-lisp/exercises/concept/conditionals/.docs/instructions.md:40]: Could you maybe explicitly mention the function name to implement (`habitat-fitter`) in this task description?

File: languages/common-lisp/exercises/concept/conditionals/.docs/instructions.md [hunk 6]
```diff
@@ -0,0 +1,88 @@
...
+
+## 3. And Now, We Feast
+
+One thing all earthling pets have in common is their need for food! This concept
+is somewhat alien to Ludwig, however, as he is prone to forgetting to refill his
+pet's food-bowl.
+
+Ludwig could use a simple function to alert him when the bowl needs
+refilling. The function would take a percent fullness (an integer) and return a
+message in the form of a string.
```
Comment (ErikSchierboom) [inline languages/common-lisp/exercises/concept/conditionals/.docs/instructions.md:62]: Could you maybe explicitly mention the function name to implement (`feeding-time-p`) in this task description?

**Review by ErikSchierboom — APPROVED** [review]
Great work! I've left some minor comments.

File: languages/common-lisp/exercises/concept/conditionals/.docs/after.md (hunk 2 above)
Comment (TheLostLambda) [reply languages/common-lisp/exercises/concept/conditionals/.docs/after.md:1]: A good catch! I couldn't find too many, but I'll transfer the ones from the hints document!

**Pull request merged**